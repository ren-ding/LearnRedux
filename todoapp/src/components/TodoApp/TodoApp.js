import React from 'react';
import AddTodo from '../AddTodo/AddTodo';
import FilterLink from '../FilterLink/FilterLink';
import VisibleTodoList from '../VisibleTodoList/VisibleTodoList';

const Footer = () => {
    return (
        <p>
            Show: 
            {' '}
            <FilterLink filter='all'>
                All
            </FilterLink>
            {' '}
            <FilterLink filter='active'>
                Active
            </FilterLink>
            {' '}
            <FilterLink filter='completed'>
                Completed
            </FilterLink>
        </p>
    );
};

const TodoApp  = () => (
    <div>
        <AddTodo />
        <VisibleTodoList />
        <Footer />
    </div>
);

export default TodoApp;