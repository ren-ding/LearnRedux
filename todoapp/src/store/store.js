import {createStore} from 'redux'
import todoApp from '../reducer/reducer'

// const persistedState = {
//     todos: [{
//         id:'0',
//         text:'Welcome back',
//         completed: false
//     }]
// };

const store = createStore(
    todoApp,
    //persistedState
);

export default store;