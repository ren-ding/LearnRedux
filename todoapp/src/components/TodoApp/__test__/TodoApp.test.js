import React from 'react';
import renderer from 'react-test-renderer';
import TodoApp from '../TodoApp';
import store from '../../../store/store';
import {Provider} from 'react-redux';

describe('TodoApp',()=>{
    let props;

    beforeEach(()=>{
        props = {
            store: store
        }
    });

    describe('render',()=>{
        it('should render match with snapshot',()=> {
            const tree = renderer.create(
                <Provider {...props}>
                    <TodoApp />
                </Provider>
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});