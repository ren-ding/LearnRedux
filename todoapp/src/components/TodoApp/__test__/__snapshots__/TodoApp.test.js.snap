// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoApp render should render match with snapshot 1`] = `
<div>
  <div>
    <input />
    <button
      onClick={[Function]}
    >
      Add Todo
    </button>
  </div>
  <p>
    Loading...
  </p>
  <p>
    Show:
     
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={
        Object {
          "color": "black",
          "textDecoration": "none",
        }
      }
    >
      All
    </a>
     
    <a
      aria-current={null}
      href="/active"
      onClick={[Function]}
    >
      Active
    </a>
     
    <a
      aria-current={null}
      href="/completed"
      onClick={[Function]}
    >
      Completed
    </a>
  </p>
</div>
`;
